File: API/components/udp_client.ts
import { udpServer } from "../api"
import { ServerMessage } from "../data/server_message"

class UDPClient {
    readonly port: number
    readonly address: string

    constructor(address: string, port: number) {
        this.port = port
        this.address = address
    }

    send = (udpMessage: ServerMessage) => {
        udpServer.udpServer.send(JSON.stringify(udpMessage), this.port, this.address)
    }
}

export { UDPClient }

File: GameServer/Assets/Scripts/Clients/UDPClient.cs
public class UDPClient
{
    readonly string _address;
    readonly int _port;

    public UDPClient(string address, int port) {
        _address = address;
        _port = port;
    }

    public string Address => _address;

    public int Port => _port;

    public void Send<T>(T message) {
        API.UdpServer.Send(this, message);
    }
}
